File: CS/WindowsApplication7/TabbedMDI.cs
using System.Windows.Forms;
using DevExpress.Utils.Menu;
using DevExpress.XtraEditors;

namespace WindowsApplication7 {
    public partial class frmTabbedMDI : DevExpress.XtraEditors.XtraForm {
        MenuHelper helper;

        public frmTabbedMDI() {
            InitializeComponent();
            for(int i = 0; i < 3; i++) AddNewForm();
            helper = new MenuHelper(xtraTabbedMdiManager1);
            helper.PopupMenuShowing += OnTabHeaderPopupMenuShowing;
        }
        //NEW EVENT
        void OnTabHeaderPopupMenuShowing(object sender, PopupMenuShowingEventArgs e) {
            if(e.TabPage.Text.EndsWith("0"))
                e.Menu.Items.Add(new DXMenuItem("Item 0"));
            if(e.TabPage.Text.EndsWith("1"))
                e.Menu.Items.Add(new DXMenuItem("Item 1"));
            if(e.TabPage.Text.EndsWith("2"))
                e.Menu.Items.Add(new DXMenuItem("Item 2"));
        }
        int formCount = 0;
        void AddNewForm() {
            XtraForm frm = new XtraForm();
            frm.Text = string.Format("Form {0}", formCount++);
            frm.MdiParent = this;
            frm.Show();
        }
        void xtraTabbedMdiManager1_PageAdded(object sender, DevExpress.XtraTabbedMdi.MdiTabPageEventArgs e) {
            e.Page.Image = imageList1.Images[formCount % imageList1.Images.Count];
        }
        protected override void OnFormClosing(FormClosingEventArgs e) {
            base.OnFormClosing(e);
            helper.PopupMenuShowing -= OnTabHeaderPopupMenuShowing;
        }
    }
}

File: VB/WindowsApplication7/TabbedMDI.vb
Imports Microsoft.VisualBasic
Imports System.Windows.Forms
Imports DevExpress.Utils.Menu
Imports DevExpress.XtraEditors

Namespace WindowsApplication7
	Partial Public Class frmTabbedMDI
		Inherits DevExpress.XtraEditors.XtraForm
		Private helper As MenuHelper

		Public Sub New()
			InitializeComponent()
			For i As Integer = 0 To 2
				AddNewForm()
			Next i
			helper = New MenuHelper(xtraTabbedMdiManager1)
			AddHandler helper.PopupMenuShowing, AddressOf OnTabHeaderPopupMenuShowing
		End Sub
		'NEW EVENT
		Private Sub OnTabHeaderPopupMenuShowing(ByVal sender As Object, ByVal e As PopupMenuShowingEventArgs)
			If e.TabPage.Text.EndsWith("0") Then
				e.Menu.Items.Add(New DXMenuItem("Item 0"))
			End If
			If e.TabPage.Text.EndsWith("1") Then
				e.Menu.Items.Add(New DXMenuItem("Item 1"))
			End If
			If e.TabPage.Text.EndsWith("2") Then
				e.Menu.Items.Add(New DXMenuItem("Item 2"))
			End If
		End Sub
		Private formCount As Integer = 0
		Private Sub AddNewForm()
			Dim frm As New XtraForm()
			frm.Text = String.Format("Form {0}", formCount)
			formCount += 1
			frm.MdiParent = Me
			frm.Show()
		End Sub
		Private Sub xtraTabbedMdiManager1_PageAdded(ByVal sender As Object, ByVal e As DevExpress.XtraTabbedMdi.MdiTabPageEventArgs) Handles xtraTabbedMdiManager1.PageAdded
			e.Page.Image = imageList1.Images(formCount Mod imageList1.Images.Count)
		End Sub
		Protected Overrides Sub OnFormClosing(ByVal e As FormClosingEventArgs)
			MyBase.OnFormClosing(e)
			RemoveHandler helper.PopupMenuShowing, AddressOf OnTabHeaderPopupMenuShowing
		End Sub
	End Class
End Namespace